cmd/kgctl/graph.go

```diff
@@ -38,6 +39,20 @@ func runGraph(_ *cobra.Command, _ []string) error {
 	if err != nil {
 		return fmt.Errorf("failed to list peers: %v", err)
 	}
+	// Optain the Granularity by looking at the annotation of the first node.
+	if opts.granularity == mesh.AutoGranularity {
```

Comment:
How about we turn this block into a helper since we seem to have the same logic repeated 3 times?